Ubuntu installer

ingoratsdorf opened the pull request:
Adds bare metal installer for ubuntu. Tested with version 24.04. You may want to or have to change the PHPVERSION variable in the start script for other versions

jokob-sk commented:
Thanks a lot for the PR @ingoratsdorf 🙏 
Feel free to cleanup the debian install if you'd like... I was also thinking how to have pip and install dependencies in one file instead of directly in separate scripts - if you are in a refactoring mood :) 

 otherwise, let me know once this PR is good to merge if you want to address the coderabbit comments first - thanks a lot

ingoratsdorf commented:
> Thanks a lot for the PR @ingoratsdorf 🙏 Feel free to cleanup the debian install if you'd like... I was also thinking how to have pip and install dependencies in one file instead of directly in separate scripts - if you are in a refactoring mood :)
> 
> otherwise, let me know once this PR is good to merge if you want to address the coderabbit comments first - thanks a lot

I used your Debian installer as base, but had to fix a few things for Ubuntu.
There's no RC system on Ubuntu, so some plugins like avahi throw errors as they use RC to check for running services. TODO in plugins. No drama at this stage.
There is no PHP8.2 in the current Ubuntu 24.04, only 8.3. 8.2 is deprecated. Dunno about Debian, guess that may (or not) depend on the version.

Added a few things for robustness and a few convenience things like choice between delete install dir or just run an update or just nothing and start. Especially the whole Nginx startup, links, config files created a lot of issues when anything old was there.

Should all work now, I ran it a few times with all sorts of scenarios.
Added tmpfs for api and log.

jokob-sk commented:
Great! 🎊Can you please update the docs too or do you want to submit a separate pr?
